src/black/mode.py

```diff
@@ -136,13 +143,23 @@ class Mode:
     experimental_string_processing: bool = False
     preview: bool = False
 
+    def __post_init__(self) -> None:
+        if self.experimental_string_processing:
+            warn(
+                "`experimental string processing` has been included in `preview`"
+                " and deprecated. Use `preview` instead.",
+                Deprecated,
```

Comment:
The reason I'm using `warnings.warn` and warning *here*, is that if we define the public API soon, then the warning is issued when `Mode` is instantiated. But having it in CLI main works as well.